lib/binding_web/binding.js

```diff
@@ -1280,6 +1280,13 @@ class Query {
     if (typeof matchLimit !== 'number') {
       throw new Error('Arguments must be numbers');
     }
+    if (endIndex != 0 && startIndex > endIndex) {
+      throw new Error('startIndex cannot be greater than endIndex');
+    }
+    if (endPosition != ZERO_POINT && (startPosition.row > endPosition.row ||
```

Comment:
Logic here is duplicated below. Wasn't sure if it should be pulled into a helper/ where that would live in the file.

Reply:
I think it's fine